log4j-core/src/main/java/org/apache/logging/log4j/core/net/ssl/SslConfiguration.java

```diff
@@ -181,61 +123,43 @@ private static SSLContext createDefaultSslContext(final String protocol) {
 
     private static SSLContext createSslContext(
             final String protocol,
-            final KeyStoreConfiguration keyStoreConfig,
-            final TrustStoreConfiguration trustStoreConfig)
-            throws KeyStoreConfigurationException, TrustStoreConfigurationException {
+            @Nullable final KeyStoreConfiguration keyStoreConfig,
+            @Nullable final TrustStoreConfiguration trustStoreConfig) {
         try {
-            KeyManager[] kManagers = null;
-            TrustManager[] tManagers = null;
-
-            final SSLContext newSslContext = SSLContext.getInstance(protocol);
-            if (keyStoreConfig != null) {
-                final KeyManagerFactory kmFactory = loadKeyManagerFactory(keyStoreConfig);
-                kManagers = kmFactory.getKeyManagers();
-            }
-            if (trustStoreConfig != null) {
-                final TrustManagerFactory tmFactory = loadTrustManagerFactory(trustStoreConfig);
-                tManagers = tmFactory.getTrustManagers();
-            }
-
-            newSslContext.init(kManagers, tManagers, null);
-            return newSslContext;
-        } catch (final NoSuchAlgorithmException e) {
-            LOGGER.error("No Provider supports a TrustManagerFactorySpi implementation for the specified protocol", e);
-            throw new TrustStoreConfigurationException(e);
-        } catch (final KeyManagementException e) {
-            LOGGER.error("Failed to initialize the SSLContext", e);
-            throw new KeyStoreConfigurationException(e);
+            final SSLContext sslContext = SSLContext.getInstance(protocol);
+            final KeyManager[] keyManagers = loadKeyManagers(keyStoreConfig);
+            final TrustManager[] trustManagers = loadTrustManagers(trustStoreConfig);
+            sslContext.init(keyManagers, trustManagers, null);
+            return sslContext;
+        } catch (final Exception error) {
+            LOGGER.error(
+                    "Failed to create an `SSLContext` using the provided configuration, falling back to a default instance",
+                    error);
+            return createDefaultSslContext(protocol);
```

Comment:
Not sure if this is the correct default.

The default `SSLContext` trusts more than a 100 public Certification Authorities. I always have a problem with such a trust store. Maybe the "empty" context is a better fallback? It will certainly cause a failure that the user will detect.

Reply:
I don't really have a particular preference. Wouldn't providing an empty trust store here break the system of users configuring only a key store, but leaving the trust store configuration out? That is, if we provide an empty trust store by default, such systems won't be able to connect to hosts they were earlier able to. What do you think?

Reply:
I checked the implementation and:

- Passing `null` as `TrustManager[]` parameter uses the **default** trust store.
- Passing `null` as `KeyManager[]` parameter uses an **empty** key store (not the same as the key store specified through `javax.net.ssl.*` properties).

So users have 3 choices:

1. Use `javax.net.ssl.*` properties and don't specify anything in Log4j.
2. Specify just a `KeyStore` and rely on the **default** `TrustStore`.
3. Specify both a `KeyStore` and a `TrustStore`.

If a user wants to use (2) or (3), but he misconfigures the system, I am not sure he wants to fall back to (1). I am not strongly opinionated, if you believe (1) is the correct fall back, we can keep it this way.

Reply:
Thanks for the investigation, much appreciated. I am more concerned about breaking working systems, hence, I will keep the existing behaviour. Maybe we can revisit this suggestion later.